lib/eventsource.js

```diff
@@ -146,19 +148,15 @@ function EventSource (url, eventSourceInitDict) {
         return
       }
 
-      // Handle HTTP redirects
-      if (res.statusCode === 301 || res.statusCode === 302 || res.statusCode === 307) {
-        if (!res.headers.location) {
-          // Server sent redirect response without Location header.
-          _emit('error', new Event('error', {status: res.statusCode, message: res.statusMessage}))
```

Comment:
I think this isn't important, but we are loosing this error reporting. Probably fine?